Monads.agda

```diff
@@ -23,38 +23,38 @@ record MathMon (M : Set → Set) : Set₁ where
     unit : {A : Set} → A → M A
     mult : {A : Set} → M (M A) → M A
     -- functoriality
-    fun-composition : {A B C : Set} → {f : B → C} → {g : A → B} → fmap (f ∘ g) ≡ (fmap f) ∘ (fmap g)
-    fun-identity : {A : Set} → fmap {A} id ≡ id
+    fun-composition : ∀ {A B C : Set} → ∀ {f : B → C} → ∀ {g : A → B} → fmap (f ∘ g) ≡ (fmap f) ∘ (fmap g)
+    fun-identity : ∀ {A : Set} → fmap {A} id ≡ id
     -- naturality
-    nat-unit : {A B : Set} → {f : A → B} → (fmap f) ∘ unit ≡ unit ∘ f
-    nat-comp : {A B : Set} → {f : A → B} → (fmap f) ∘ mult ≡ mult ∘ fmap (fmap f)
+    nat-unit : ∀ {A B : Set} → ∀ {f : A → B} → (fmap f) ∘ unit ≡ unit ∘ f
+    nat-comp : ∀ {A B : Set} → ∀ {f : A → B} → (fmap f) ∘ mult ≡ mult ∘ fmap (fmap f)
     -- consistency
-    con-unit₁ : {A : Set} → mult {A} ∘ fmap unit ≡ id
-    con-unit₂ : {A : Set} → mult {A} ∘ unit ≡ id
-    con-mult  : {A : Set} → mult {A} ∘ fmap mult ≡ mult ∘ mult
+    con-unit₁ : ∀ {A : Set} → mult {A} ∘ fmap unit ≡ id
+    con-unit₂ : ∀ {A : Set} → mult {A} ∘ unit ≡ id
+    con-mult  : ∀ {A : Set} → mult {A} ∘ fmap mult ≡ mult ∘ mult
 
 record ProgMon (M : Set → Set) : Set₁ where
   field
     unit : {A : Set} → A -> M A
     _>>=_ : {A B : Set} → M A → (A → M B) -> M B
     -- monadicity
-    unitˡ : {A B : Set} → ∀ {x : A} → {f : A → M B}
+    unitˡ : ∀ {A B : Set} → ∀ {x : A} → {f : A → M B}
       → (unit x) >>= f ≡ f x
-    unitʳ : {A : Set} → ∀ {m : M A}
+    unitʳ : ∀ {A : Set} → ∀ {m : M A}
       → m >>= unit ≡ m
-    assoc : {A B C : Set} → ∀ {m : M A} → {f : A → M B} → {g : B → M C}
+    assoc : ∀ {A B C : Set} → ∀ {x : A} → ∀ {f : A → M B} → ∀ {g : B → M C} → ∀ {m : M A}
```

Comment:
`x` is unbound (and a typo). We need signature like (note the order, which is assumed in the proofs)
```
assoc : {A B C : Set} → ∀ {m : M A} → ∀ {f : A → M B} → ∀ {g : B → M C}
  → (m >>= f) >>= g ≡ m >>= λ{x → f x >>= g}

```